cmd/csi-resizer/main.go

```diff
@@ -156,7 +171,8 @@ func main() {
 		*timeout,
 		kubeClient,
 		informerFactory,
-		driverName)
+		driverName,
+		*enableNodeDeployment)
```

Comment:
There is one potential scalability problem here: if there are now as many resizers running in the cluster as there are nodes, then the apiserver has to keep all of them informed about PV and PVC updates.

external-provisioner has the same problem, but it cannot really be avoided there because support for immediate provisioning depends on all provisioners seeing a new PVC.

I'm just wondering whether something better can be done for the other sidecars. The code below already checks for the `util.VolumeSelectedNodeKey` label. It would be possible to set up the PVC informer so that it does server-side filtering. It's doable, I just don't have a code example at hand.

That leaves the PV, though. external-provisioner would have to be modified to set a similar (the same?) label there.




Perhaps external-provisioner can set a special label for

Reply:
Here is code which sets up an informer with server-side filtering:
https://github.com/kubernetes-csi/external-provisioner/blob/3b752c36ca71fbf5d7310bb6a568b93844062189/cmd/csi-provisioner/csi-provisioner.go#L452-L464

Reply:
I thought about that as well but didn't want to 'kick against the pricks' as they say with how the others were implemented. I also noted that we are not currently doing server-side filtering on the driver either (which I found odd).

Given my limited `golang` skill set I would definitely need hand-holding on proper code to manage the watches/filters should we go that route.